### PR TITLE
[aggregator] Minor improvements

### PR DESCRIPTION
**Pull request by georgemitenkov**
Refactoring of aggregators V1 and V2. No new logic added.

1. We now how a dedicated `test` folder. So far only
    test utilities are there, e.g. fake aggregator view.
2. Aggregators use `VersionedID` which is either a 
    state key (V1) or an int (V2). Previously this separation
    was a bit vague.
3. Some minor refactoring included: names of functions, 
    removed unused types / APIs.

### CODE REVIEW

**Review by igor-aptos — COMMENTED** [review]
Overall looks good.

What is the reason for pulling bounded math unit tests to a separate folder?

I find it cleaner to have real unit tests (i.e. things that don't depend on anything else, not integration tests) to be in the same file as is the code they are testing. They are also part of the documentation of what is the expected behavior (another example from today - https://github.com/aptos-labs/aptos-core/pull/10072/files ).

Unless you have a good reason to move them out, can you revert that change? the rest looks good

**Comment by igor-aptos** [comment]
also don't refactor more things before few of the PRs touching extension/native code land :)

**Comment by georgemitenkov** [comment]
@igor-aptos for tests the motivation was to avoid the code from growing too large, e.g. in extension, but let's do it if we really need then. Will revert that part 👍

**Review by gelash — APPROVED** [review]

**Review by igor-aptos — APPROVED** [review]

File: aptos-move/aptos-aggregator/src/aggregator_extension.rs [hunk 1]
```diff
@@ -436,12 +390,12 @@ pub struct AggregatorData {
...
+    new_aggregators: BTreeSet<AggregatorVersionedID>,
     // All aggregators that were destroyed in the current transaction, stored as ids.
-    destroyed_aggregators: BTreeSet<AggregatorID>,
+    destroyed_aggregators: BTreeSet<StateKey>,
     // All aggregator instances that exist in the current transaction.
-    aggregators: BTreeMap<AggregatorID, Aggregator>,
-    // All aggregatorsnapshot instances that exist in the current transaction.
+    aggregators: BTreeMap<AggregatorVersionedID, Aggregator>,
+    // All aggregator snapshot instances that exist in the current transaction.
     aggregator_snapshots: BTreeMap<AggregatorID, AggregatorSnapshot>,
```
[the remark this answers is not in the document]
Comment (georgemitenkov) [reply aptos-move/aptos-aggregator/src/aggregator_extension.rs:399]: Yep if needed we can change it back to versioned

**Pull request merged**